### ISSUE TITLE
How to add some props to all components in router?

### ISSUE THREAD

**Issue by hhashmi** [sysgears/apollo-fullstack-starter-kit#160]
How to add some props to all components in router?
if there is any way that I can inject these props even to all the nested components. Want to use these like GLOBAL PROPS for all the components in application

In earlier version of start kit I did

**in src/ui/component/app.jsx**

```
render() {
    var childrenWithProps = React.cloneElement(this.props.children, {'socketConn': socket, 'timeClock':this.state.timeClock, 'applicationEnv': ApiUtils.getEnvironment(), 'applicationStatus': this.state.applicationStatus, 'applicationOnline': this.state.applicationOnline});
    return (
      <div>
        <NavBar>
        </NavBar>
        <TimeClock timeClock={this.state.timeClock} applicationStatus={this.state.applicationStatus} applicationOnline={this.state.applicationOnline} />
        <div className="container">
          {childrenWithProps}
        </div>
        <footer className={classnames('container', css(styles.footer))}>
          <Row className='text-center'>
            &copy; 2017. My App.
          </Row>
        </footer>
      </div>
    );
  }
```

**Comment by vlasenko** [comment]
@hhashmi Consider storing global state in `Redux` store. See how `redux` counter example is implemented. Consider using `Redux` middleware for working with socket connection.

**Comment by hhashmi** [comment]
I'm using this socket connection to get accurate and continues server time to display a digital clock in my application. 

Will redux store be ok for this purpose ?

**Comment by vlasenko** [comment]
Yes, of course. For integrating sockets into the redux store you might want to use the approach like this:
http://stackoverflow.com/questions/42330094/where-to-store-socket-connection-in-read-redux

Generally, all the global state of the application on client-side should be kept in Redux store.

**Comment by mitjade** [comment]
@hhashmi I needed to do something similar and just done it with graphQL and subscriptions. Created a query `currentDateTime` and then used `setInterval` to send it via subscriptions.

Not sure if this is a better approach, but did not have to setup a lot of things to get this running.

**Comment by hhashmi** [comment]
getting the value for displaying is not an issue for me. It is working fine. I just made a component and in app.jsx I connected to socket and getting the values.
Now my next task if to inject that time value I'm getting from server to all of my components following my router.

I need to make decisions in most of components based on this time value like expire values,  activate jobs, mark delayed jobs and etc.

so my question is simple how can I inject that value to other components, just like I did in earlier version of starter kit.
